src/bit-systems/media-loading.ts

```diff
@@ -99,6 +100,9 @@ function resizeAndRecenter(world: HubsWorld, media: EntityID, eid: EntityID) {
   const mediaObj = world.eid2obj.get(media)!;
   const box = new Box3();
   box.setFromObject(mediaObj);
+  if (box.isEmpty()) {
+    box.setFromCenterAndSize(new Vector3(0, 0, 0), new Vector3(1, 1, 1));
+  }
```

Comment:
Wondering in what scenarios box can be empty...

If I'm right, aren't visible objects always assigned to media entities by Hubs Client (media loaders) for loaded medias (eg: Plane Mesh for videos and audios)?

Reply:
For example when a glTF is loaded that has a media component attached, when `resizeAndRecenter` is called at that point there is no actual geometry in the gltf hierarchy as the video plane is added when the video is resolved and so the AABB is not set. As we resize and recenter the media based on that size that causes the media to be positioned at (0,0,0).

This will happen with any gltf that has a component that triggers an async media fetch: model, video, audio, image and link.

Reply:
A better solution would be to resize and center when all the depending medias have been resolved but we don't have a good way of knowing that so that would be a more complicated solution.

Reply:
> For example when a glTF is loaded that has a media component attached, when resizeAndRecenter is called at that point there is no actual geometry in the gltf hierarchy as the video plane is added when the video is resolved

Isn't this `resizeAndRecenter()` called after video (or other media) is resolved?

https://github.com/mozilla/hubs/blob/c5b17010a8aa2e7522ed3506e3c7e2a621d3f344/src/bit-systems/media-loading.ts#L213-L216

Reply:
Yes but:
- We only recenter the spawned interactable not the media:
Interactable:
https://github.com/mozilla/hubs/blob/c5b17010a8aa2e7522ed3506e3c7e2a621d3f344/src/load-media-on-paste-or-drop.ts#L26-L32
Media (video example):
https://github.com/mozilla/hubs/blob/c5b17010a8aa2e7522ed3506e3c7e2a621d3f344/src/inflators/video-loader.ts#L14-L20
- Even if we recenter the child media, that would only recenter it on the parent not the scene.

Reply:
Ah, I had misread this part

> This will happen with any gltf that has a component that triggers an async media fetch: model, video, audio, image and link.

Would you mind adding a comment about it? Other people may be confused like I did. Or I would like to add a comment in another PR.